cpu/fe310/include/context_frame.h

```diff
@@ -0,0 +1,129 @@
+
+/*
+ * Copyright (C) 2017 JP Bonn
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    cpu_fe310
+ * @ingroup     cpu
+ * @brief       Freedom E cpu
+ * @{
+ *
+ * @file
+ * @brief       Thread context frame stored on stack.
+ *
+ * @author      JP Bonn
+ */
+
+#ifndef CONTEXT_FRAME_H
+#define CONTEXT_FRAME_H
+
+#if !defined(__ASSEMBLER__)
+#include <stdint.h>
+#include <assert.h>
+#endif /* __ASSEMBLER__ */
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   context_switch_frame stores the registers and PC for a context switch.
+ *
+ * This also defines context_switch_frame offsets for assembly language.  The
+ * structure is sized to maintain 16 byte stack alignment per the ABI.
+ * https://github.com/riscv/riscv-elf-psabi-doc
+ *
+ */
+#if !defined(__ASSEMBLER__)
+
+/* N.B.: update the definitions below if this changes */
+struct context_switch_frame {
+    uint32_t filler0;   /* filler to maintain 16 byte alignment */
+    uint32_t filler1;   /* filler to maintain 16 byte alignment */
+    uint32_t pc;
+    /* Callee saved registers */
+    uint32_t s0;
+    uint32_t s1;
+    uint32_t s2;
+    uint32_t s3;
+    uint32_t s4;
+    uint32_t s5;
+    uint32_t s6;
+    uint32_t s7;
+    uint32_t s8;
+    uint32_t s9;
+    uint32_t s10;
+    uint32_t s11;
+    /* Caller saved register */
+    uint32_t ra;
+    uint32_t tp;
+    uint32_t t0;
+    uint32_t t1;
+    uint32_t t2;
+    uint32_t t3;
+    uint32_t t4;
+    uint32_t t5;
+    uint32_t t6;
+    uint32_t a0;
+    uint32_t a1;
+    uint32_t a2;
+    uint32_t a3;
+    uint32_t a4;
+    uint32_t a5;
+    uint32_t a6;
+    uint32_t a7;
+};
+#endif /* __ASSEMBLER__ */
+
+
+/* These values are checked for correctness in context_frame.c */
```

Comment:
Add some general docs for what these macros are. You could do a `/* @name something @{ `